app/helpers/application_helper.rb

```diff
@@ -42,6 +42,17 @@ def bookmarklet_code(height=400, width=620)
       "bookmarklet('#{bookmarklet_url}', #{width}, #{height});"
   end
 
+  def show_local_posts_link?(current_user)
+    return true if AppConfig.settings.enable_show_local_post_link == "always"
+    return true if AppConfig.settings.enable_show_local_post_link == "admin" &&
+                   Role.is_admin?(current_user)
+
+    return true if AppConfig.settings.enable_show_local_post_link == "moderator" &&
+                   (Role.moderator?(current_user) || Role.is_admin?(current_user))
+
+    false
+  end
```

Comment:
How about moving this to `lib/configuration_methods.rb` like so

```rb
def has_local_posts_stream?(user)
    return true if settings.enable_local_posts_stream == "admins" &&
                   Role.is_admin?(user)
    return true if settings.enable_local_posts_stream == "moderators" &&
                   (Role.moderator?(user) || Role.is_admin?(user))

    settings.enable_local_posts_stream == "everyone"
end
```

Then we can reuse the logic in the controller.

Reply:
Great Idea - I did not know this class until now.